components/engine/Cargo.toml

```diff
@@ -8,28 +8,15 @@ publish = false
 jemalloc = ["rocksdb/jemalloc"]
 portable = ["rocksdb/portable"]
 sse = ["rocksdb/sse"]
-protobuf-codec = [
-  "kvproto/protobuf-codec",
-  "raft/protobuf-codec",
-]
-prost-codec = [
-  "prost",
-  "kvproto/prost-codec",
-  "raft/prost-codec",
-]
 
 [dependencies]
 configuration = { path = "../configuration" }
 encryption = { path = "../encryption" }
 engine_traits = { path = "../engine_traits" }
 hex = "0.4"
-kvproto = { git = "https://github.com/pingcap/kvproto.git", default-features = false }
 lazy_static = "1.3"
 prometheus = { version = "0.8", features = ["nightly", "push", "process"] }
-prost = { version = "0.6", optional = true }
 protobuf = "2.8"
-quick-error = "1.2.2"
-raft = { version = "0.6.0-alpha", default-features = false }
```

Comment:
So the dependency is not used at all in the past?

Reply:
It was. There was a `From<engine::Error> for raft::Error` conversion. Now that conversion is done without depending on `raft` from `engine_trait` through the tiny `into_other` crate.